src/emoji/data.js

```diff
@@ -20,6 +21,16 @@ export const codeToEmojiMap: {| [string]: string |} = objectFromEntries<string,
   }),
 );
 
+// TODO(?): Stop having distinct `EmojiType` and `ReactionType`; confusing?
```

Comment:
Yeah, quite possibly. It'd be nice to unify these.

The thing this was solving is that a Zulip extra emoji is, by design, really a *lot* like a realm emoji. They're different in that the Zulip extra emoji is always there and you can't edit it, and in that it's internally named slightly differently and you find its URL in a different way… but then both of them are just an image that has some URL on the server. And when we first added support for the Zulip extra emoji, it looked like we were going to duplicate a bunch of code to do it, so this was a way of not doing that. Original discussion here: https://github.com/zulip/zulip-mobile/pull/2375#issuecomment-450101731

I think it should be possible to refactor this so that we don't have this `EmojiType` competing with `ReactionType` and also don't have the code that's dealing with emoji get a bunch more complex or duplicated. Probably a key tactic in doing so would be:
* Instead of having the rest of the app pass around things like `allImageEmojiById: $ReadOnly<{| [id: string]: ImageEmojiType |}>` and expect to know how to look emoji up in that data, have some kind of `EmojiState` type.
* Take the places where we look inside one of those object-as-map values, and move those to be encapsulated in an emoji model: the rest of the app just passes around `EmojiState`, and passes that to the emoji model, along with reaction_type and emoji_code, to look up the details it needs to know.
* OTOH that's still going to end up returning something that's bifurcated between image emoji and Unicode emoji, because that's fundamentally the shape of the information that the UI needs in order to display the emoji. See `Emoji.js`, and `messageReactionAsHtml` in `html/message.js`.
* But it might be a union of object types, looking like `{| type: 'unicode', text: string |} | {| type: 'image', url: string |}`.

This is probably more refactoring than is worth doing right now -- I don't think we have a ton more emoji-related changes to make right now. But that's the thoughts I have from spending a little time looking into this area just now.